src/main/java/com/example/Trello/services/impl/CardServiceImpl.java

```diff
@@ -0,0 +1,79 @@
+package com.example.Trello.services.impl;
+
+import com.example.Trello.mappers.CardMapper;
+import com.example.Trello.model.dto.board.Board;
+import com.example.Trello.model.dto.card.Card;
+import com.example.Trello.model.dto.card.CardCreation;
+import com.example.Trello.repositories.BoardRepository;
+import com.example.Trello.repositories.CardRepository;
+import com.example.Trello.services.BoardService;
+import com.example.Trello.services.CardService;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+import java.util.Optional;
+
+@Service
+@Slf4j
+public class CardServiceImpl implements CardService {
+
+    private final CardMapper cardMapper;
+
+    private final CardRepository cardRepository;
+
+    private final BoardService boardService;
+
+    public CardServiceImpl(CardMapper cardMapper, CardRepository cardRepository, BoardService boardService) {
+        this.cardMapper = cardMapper;
+        this.cardRepository = cardRepository;
+        this.boardService = boardService;
+    }
+
+    @Override
+    public void addCard(final long id, final CardCreation cardCreation) {
+        Card card = cardMapper.map(cardCreation);
+        card.setBoard(boardService.getBoardById(id));
+        cardRepository.save(card);
+    }
+
+    @Override
+    public List<Card> getCards(final long id) {
+        return boardService.getBoardById(id).getCards();
```

Comment:
In fact, you do 2 queries here:
1) to get board
2) to get cards
You don't need board info at all. Use cardRepositoty to query cards by board I'd